src/main/java/org/opensearch/knn/common/TaskRunner.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.common;
+
+import org.opensearch.client.Client;
+import org.opensearch.common.util.concurrent.ThreadContext;
+
+import java.util.function.Supplier;
+
+/**
+ * Class abstracts execution of runnable or function in specific context
+ */
+public class TaskRunner {
```

Comment:
can we rename this class to ThreadContextHelper?

Reply:
yes, this makes more sense. I could came up with a good name yesterday so took the straightforward one. I'll push a commit soon